src/loader/mod.rs

```diff
@@ -211,20 +215,25 @@ pub fn load_cmdline<M: GuestMemory>(
     guest_addr: GuestAddress,
     cmdline: &Cmdline,
 ) -> Result<()> {
-    let len = cmdline.as_str().len();
-    if len == 0 {
-        return Ok(());
-    }
+    // We need to get a vector of bytes representing the cmdline
```

Comment:
This comment is no longer accurate, maybe we can rephrase it to something like: "We need a null terminated string because that's what the linux kernel expects when parsing the command line".